### PR TITLE
Add types definitions for Scaleway serverless functions runtime

### PR DESCRIPTION
**Pull request by MrMicky-FR**
This PR adds types definitions for [Scaleway serverless functions](https://www.scaleway.com/en/serverless-functions/) runtime (not an NPM package). These types are based on the AWS Lambda ones, but as only some properties are implemented I find it's easier to have only the existing types during development (rather than unexpected errors during testing). I used the implementation details available on the [GitHub repository of the functions runtime](https://github.com/scaleway/scaleway-functions-runtimes/blob/b1baa76eac3e13731e8f831b1c56550fda222a85/events/http.go#L70-L82).

The tests are based on the [Scaleway documentation](https://developers.scaleway.com/en/products/functions/api/).

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/scaleway-functions/tslint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "@definitelytyped/dtslint/dt.json",
+    "rules": {
+        "npm-naming": false
```
Comment (gabritto) [inline types/scaleway-functions/tslint.json:4]: I don't think you need to disable this rule. See comment about the header above.

File: types/scaleway-functions/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// Type definitions for scaleway-functions 1.0
```
Comment (gabritto) [inline types/scaleway-functions/index.d.ts:1]: This line should be `// Type definitions for non-npm package scaleway-functions 1.0`, to indicate that the original package is not on npm. Then with this change, I think you can re-enable the "npm-naming" rule on the tslint config below.

**Review by gabritto — CHANGES_REQUESTED** [review]

File: types/scaleway-functions/index.d.ts (hunk 2 above)
Comment (MrMicky-FR) [reply types/scaleway-functions/index.d.ts]: Thanks, I've applied the suggested changes :)

**Review by gabritto — APPROVED** [review]

**Comment by MrMicky-FR** [comment]
Ready to merge

**Pull request merged**